### ISSUE TITLE
[Video Disruptors Grant]: Waves

### ISSUE THREAD

**Issue by GageVanK** [livepeer/grants#169]
### Please describe your project. Start with the need or problem you are trying to solve with this project. Describe why your solution is going to adequately solve this problem.

Waves aims to empower content creators and live streamers with a comprehensive streaming platform that combines the power of Livepeer and the DeSo Blockchain. We seek to solve the limitations of traditional streaming platforms by providing content creators with increased control over their content, data ownership, and transparency.

### Link to public GitHub repo (if applicable)

https://github.com/GageVanK/waves-2.2

### Link to demo website (if applicable)

https://waves-2.vercel.app/

### What are the benefits to getting this right? 

**Increased Adoption and Utilization:** By integrating fully integrating Livepeer and everything it has to offer into Waves it showcases the versatility and potential of Livepeer for diverse applications in the streaming industry, encouraging more developers and businesses to adopt Livepeer for their projects.

**Deeper Integration with DeSo Blockchain:** The combination of Livepeer and Deso creates a powerful decentralized ecosystem, where content creators benefit from both Livepeer's streaming infrastructure and DeSo's blockchain for content storage and ownership.

**Showcasing Real-World Use Cases:** By offering a comprehensive streaming platform with real-world use cases for Livepeer's technology, this project serves as a tangible demonstration of the protocol's capabilities. This demonstration attracts developers, businesses, and investors to the Livepeer ecosystem, fostering growth and building confidence in the project's potential.

**Community and Developer Engagement:** The open-source nature of Waves and the emphasis on user feedback demonstrate a commitment to community engagement and collaboration. This fosters a more active and vibrant Livepeer community, encouraging developers and enthusiasts to contribute to the project's evolution.

**Contributing to Livepeer's Mission:** Livepeer's mission is to democratize access to video infrastructure and create a more open and equitable streaming ecosystem. By providing a user-friendly, decentralized, and feature-rich streaming platform, this project aligns closely with Livepeer's vision and helps drive its mission forward.

**Value to LPT Holders:** As the adoption and utilization of Livepeer's infrastructure increase through the success of this project, it positively impacts the value of the Livepeer token (LPT). More usage of Livepeer's network drives demand for LPT, benefitting existing token holders.

In summary, this proposal brings significant value to the Livepeer ecosystem by showcasing the versatility of Livepeer's infrastructure, driving adoption, fostering deeper integration with the DeSo blockchain, and contributing to the project's mission. Waves focus on community engagement and open-source development aligns perfectly with Livepeer's collaborative nature, creating a win-win scenario for all stakeholders involved.

### What are the risks if you do not get it right? 

**Limited Engagement and Activity:** Insufficient user engagement and activity on the platform could lead to a stagnant ecosystem. Viewers may not find enough content to watch, while content creators may not receive the desired audience interaction, resulting in decreased motivation to continue using the platform.

**Inability to Scale:** Scalability is crucial for a successful streaming platform, and if the project faces technical limitations in handling increasing user demands, it may struggle to grow and accommodate a growing user base.

**Wasted Resources:** A failed execution might lead to wasted time, effort, and financial resources invested in the project. This could have been used for other projects or opportunities that could have been more successful.

To mitigate these risks and execute the project successfully, thorough planning, continuous user testing and feedback, responsive bug fixing, robust security measures, and a well-defined marketing strategy are essential. Addressing technical challenges promptly, fostering a supportive community, and continuously improving the platform based on user needs will contribute to long-term success. Adherence to regulatory requirements and careful consideration of scalability and governance will further strengthen the project's viability.

### What are the risks that will make executing on this project difficult?

**Technical Challenges and Bugs:** Building a complex streaming platform with decentralized features can lead to technical challenges and potential bugs. If not addressed promptly and effectively, these issues might hinder the platform's functionality, user experience, and credibility.

**Competition:** The streaming industry is highly competitive, with established platforms like Twitch and YouTube already dominating the market as well as many new web3 streaming platforms. Introducing a new streaming platform requires compelling differentiators, user incentives, and a clear value proposition to attract users away from existing platforms.

**Protocol Limitations:** Building on a specific protocol, such as DeSo, might present limitations in terms of feature sets or governance mechanisms. Adapting the platform to align with the protocol's capabilities could be challenging and may require trade-offs.

**User Experience and Design:** The user experience and interface design are crucial for attracting and retaining users. If the platform lacks a user-friendly design and intuitive features, users might find it challenging to navigate and use effectively.

### Please describe in details what your final deliverable for this project will be.

The final deliverable for this project will be a fully functional and feature-rich streaming platform that empowers content creators and live streamers with a seamless and decentralized streaming experience. The platform will comprise several key components and features:

- **Server Side Rendering** 

- **Streamer Dashboard**

- **Record Stream Option** 

- **Video Upload Assets** 

- **Multistream**

- **In-Browser Streaming**

- **Mobile App Prototype**

- **Documentation and Demos**

- **Livepeer-Powered Banners**

- **User Feedback**

### Please break up your development work into a clear set of milestones

Right now I am the only one working on this I can definitely complete all of this within the next couple months but I am open and have been looking to partner with other developers that have built on DeSo, but for right now just me.

### Milestone 1: [$7000]
**Streamer Dashboard**
- Stream Health Status/Timer (useStream/useStreamSession)
- Recommended OBS Settings (https://docs.livepeer.org/guides/developing/stream-via-obs.en-US)
- Improve UI

 **Record Stream Option (VODs)**
- via useUpdateStream 

 **Multistream Option**
- via useUpdateStream 

**In-Browser Streaming**
- Broadcast component from LivepeerJS

**NextJS Migration + Livepeer SSR**

APIs that will use SSR:
- useAsset, useAssetMetrics, usePlaybackInfo, Player

**Video Asset Uploads**
- Upload via Livepeer (useCreateAsset) + Post via DeSo

**Livepeer Banner**
- On Home/Profile Pages

### Milestone 2: [$7000]
**Mobile App Prototype**
- Using DeSo/Livepeer's React Native Library to mirror the features of the Web App(Create Account/Login, Livestream, etc)

### Milestone 3: [$1000]

**Documentation + Demo Videos**
- All Livepeer related code/features

### The total requested budget for all milestones is $15,000.
**Developer costs: $10,000**
- Milestones 1-3

**Storage/Transcoding/Delivery Costs: $5,000**
- 1 Year worth of Livepeer's Scale Plan

### Specify your team's long-term plans to maintain this software and upgrade it over time

- Regular Updates and Bug Fixes
- Scalability and Performance Optimization
- Integration of New Features and Innovations
- User-Centric Design and Feedback
- Long-Term Community Engagement
- Partnerships and Ecosystem Growth
- DAO via [Openfund](https://openfund.com/)

### Please describe (in words) your team's relevant experience, and why you think you are the right team to build this project. You can cite your team's prior experience in similar domains, doing similar dev work, individual team members' backgrounds, etc.

Gage Van Kuilenburg
- [Portfolio](https://gages-portfolio.netlify.app/)
- [LinkedIn](https://www.linkedin.com/in/gage-v-b5a04319a/)
- Other Web3 Projects
    - [Combat Collective](https://combat-collective.netlify.app/): Social network for martial artists to connect with their fans and monetize their content 
    - [DeSo-HeroSwap](https://desoheroswap.vercel.app/): Utilized [HeroSwap](https://heroswap.com/) a no login, cross chain swapping platform and the Deso blockchain to allow users to use their DeSo account to save 50% in fees

### Who is your target user group? How do you plan on getting your users to use this?

The target user group for Waves includes current streamers and content creators who have an established presence on platforms like TikTok and Instagram Live. These creators and streamers, along with their existing audiences, are the primary focus of the platform's growth strategy.

While Waves might not have the resources to compete with multimillion-dollar deals offered by other streaming platforms (most notably Kick), it plans to attract creators through a more creative approach. The key hook for getting streamers and creators on board will be the multistreaming feature. By allowing creators to broadcast their content simultaneously on multiple platforms, Waves offers an opportunity for them to diversify their revenue streams and reach a wider audience.

As creators become more educated to the benefits of using Waves and the potential for increased revenue, they will be incentivized to make efforts to migrate their audience over to Waves. By focusing on providing practical and innovative features that cater to creators' needs and by educating them about the benefits of web3 and Waves, the platform can attract and retain users, contributing to its long-term success.

### How did you learn about the Livepeer Grants Program?

Livepeer Docs

### Was this project started at a hackathon or another web3 event? Which one?

N/A

### Please include any additional information that you think would be useful in helping us to evaluate your proposal.

After submitting the deliverables for the Microgrant, I dedicated the last month to a proactive user onboarding process, involving friends and family as initial users. This approach allowed me to gather valuable feedback from real users, leading to significant improvements in the platform. Through rigorous testing and engagement with users, I addressed various bugs and made substantial UI changes to enhance the overall user experience. This user-centric approach has been instrumental in refining the platform and ensuring it meets the needs and expectations of its target audience. I am committed to continuous improvement and will keep incorporating user feedback to deliver a robust and user-friendly streaming platform.

Waves streaming platform project sets itself apart by taking a fresh and unique approach to decentralized streaming platforms. Instead of merely creating a Twitch clone with added web3 features, Waves fully embraces text-based content alongside livestreaming, differentiating its homepage design from traditional platforms. Inspired by Twitter's content display, users can choose what content they want to see, whether through the Waves feed to find live streams or the hot feed for popular text posts. This user-centric design facilitates content discovery and creates a feedback loop for streamers to announce their upcoming streams and then easily start streaming from their dashboard.

Moreover, Waves' utilization of the DeSo protocol provides distinct advantages. Being built on a Layer 1 blockchain specifically designed to scale decentralized social platforms, DeSo offers lower content storage costs compared to other chains. Users can instantly create an account with their chosen handle and start posting on-chain, streamlining the onboarding process. While DeSo has prioritized building out its infrastructure over marketing efforts, Waves' position in the DeSo ecosystem sets it up to capitalize on the growth that comes from increased marketing exposure. Overall, Waves' proactive user onboarding, user-centric design, and strategic choice of the DeSo protocol make it a promising and unique project in the decentralized streaming space.

**Comment by hansy** [comment]
Hey @GageVanK! Good to see you here again! Thanks for applying; we'll get back to you with a decision with 2-4 weeks!

**Comment by hansy** [comment]
Hey @GageVanK, after conferring with the committee, we've decided not to fund your project at this time. We love your commitment to your project, but feel it's still a bit early for a grant. We'd like to see a little bit more traction and users before funding additional feature sets. Please continue working on Waves and keep us updated on your progress. As always, if you have any questions about Livepeer or run into any issues, please don't hesitate to reach out and we'll be more than happy to help unblock you.

**Comment by GageVanK** [comment]
Hey @hansy, is there some more specific metrics you could layout in terms of traction and users. Are there certain metrics that carry more weight than others - number of streamers, number of viewers per stream, total minutes streamed, etc? I get it's probably a combination of all those things, but it would be nice to have more clarity.

Also, in our first conversation for the microgrant you had mentioned an accelerator program near the end of the year. Is this still happening? If so, this would be my intro into accelerators and I'm just wondering is it even worth doing with where I'm at right now with waves? Does it hit the same ceiling in terms of needing more traction and for waves to mature a bit more? Sorry to bombard you with all the questions but I know you have seen many projects like waves and I just want to get super clear on what needs to happen so I can gameplan and hopefully come out of this crypto/web3 winter on top.

Thank you so much!

**Comment by hansy** [comment]
Great questions @GageVanK! Let me see if I can do my best to provide more clarity.

It's hard for us to prescribe a hard number in terms of number of streams, minutes, videos, users, etc. Every application has its own set of nuances, user base, and feature set, so we don't have any target metrics we are explicitly looking for.

However, we do want to want to see some level of growth. From our end, our most important metric is minutes transcoded, which we can monitor your application for. Again, while we don't have any specific numbers we're looking for, we'd like to see general transcoding time (whether via livestreams or videos uploaded) increase over time. At a high level, we attribute growth in transcoded minutes to growth in applications (because maybe more users are signing up and livestreaming, uploading videos, etc). Does that make sense?

In terms of the accelerator, the program is on hold. That doesn't mean we won't start a new cohort in the future; I just don't have any new details about it at the moment. Should more details emerge, I'll be sure to ping you about it, so thank you so much for your interest! Based on the selection criteria for our previous inaugural class, we more or less still look at growth, so demonstration of it is key for the program, but also for any future grants.

Please let me know if I need to elaborate on anything else. You're asking the right questions. As always, my team and I are happy to be resources for your project's continued development!
